lib/libzfs/Makefile.am

```diff
@@ -72,7 +72,7 @@ endif
 
 libzfs_la_LDFLAGS = -version-info 2:0:0
 
-libzfs_la_LIBADD += -lm $(LIBSSL)
+libzfs_la_LIBADD += -lm $(BSDLIB) $(LIBSSL)
```

Comment:
It would be nice if we could add this functionality without bringing in a new dependency on `bsdlib`.  The library seems to be widely available on the major Linux platforms, but depending on it in the `zfs` utility does mean we'll need to make sure it's available even in the initramfs when booting.  There are some other small gotchas as well involving updating the packaging, and maybe others. That's workable but perhaps not ideal.

The other consideration is that from a design standpoint we would now be using freebsd wrapper functions from two different sources.  The `libbsd` library and the zfs provided `libspl` library.  That's going to make things a bit more confusing.  I'd really prefer that we pick a single library to provide the needed functionality.

Skimming it, it looks to largely be a wrapper around `prctl(2)` with a little slight of hand to allow for longer names, so perhaps adding it wouldn't be too bad.  It appears to be under the Expat license as well which is compatible.  Thoughts?

Reply:
I don't see how it's a wrapper arond prctl(2).  I looked at a couple of implementations of setproctitle() and none of them seem to use it.

I'm perfectly happy to simplify, and cheerfully admit Linux isn't my strongest platform.  So I'm happy to accept suggestions.

Reply:
Nether do I after taking a closer second look!   Sorry about that.

I think a reasonable way to simplify this would be to add a platform specific version of `setproctitle` in `lib//libspl/os/linux/setproctitle.c` for Linux.  It appears multiple projects have added their own kludgy implementation over the years since Linux kernel still doesn't provide support for it.  For example, even the util-linux package seems to have added the version from sendmail, https://github.com/karelzak/util-linux/blob/master/lib/setproctitle.c.  That should be fine for this.  For the moment, I'd just like to avoid the additional dependency.

When you get a chance would you also mind rebasing this PR on the master branch and squashing your commits.  It's a big difficult to review at the moment with the merge commit in the PR.  You will need to force update the PR, but that's fine.  Thanks!

Reply:
It also needs a constructor hook, to initialize the arguments.  (Or put in a call to initproctitle in every command.)

Reply:
OK.  I did a little more research and it seems `libbsd` is already commonly included in initramfs images.  So this may not be a major issue, and we can always leave it disabled on Linux as needed since `libbsd` is still optional.  That said, I do think it would be worth looking in to using some of the other functionality it provides so we don't have two libraries providing the same functions.